bash_completion

```diff
@@ -1010,6 +1024,8 @@ _init_completion()
                 ;;
         esac
     done
+    shift "$((OPTIND - 1))"
+    (($#)) && comp_args=("$@")
```

Comment:
Would it make sense to do this only if `comp_args` is defined? Otherwise I suppose completions not prepared for the new behavior will leak a global by that name unless I'm missing something?

```suggestion
    if [[ -v comp_args ]]; then
        shift "$((OPTIND - 1))"
        (($#)) && comp_args=("$@")
    fi
```

Reply:
`local ... comp_arg` doesn't make the variable *set state*. For `[[ -v comp_args ]]` to success, an actual value should be assigned to `comp_arg` prior to the test. So,

1. one way is to request writing like `local ... comp_arg=`.
2. Another way is to test the existence of the variable by `declare comp_args &>/dev/null`.

But I'm not sure if we need to test it explicitly because I think we can assume that `comp_args` is declared when extra arguments are specified to `_comp_initialize` or `_init_completion`. This is tested by `(($#)) &&`.